components/scream/src/share/grid/remap/abstract_remapper.hpp

```diff
@@ -143,15 +161,37 @@ class AbstractRemapper
   virtual const field_type& do_get_src_field (const int ifield) const = 0;
   virtual const field_type& do_get_tgt_field (const int ifield) const = 0;
 
+  // Override this method to insert logic executed at the beginning of the field
+  // registration process.
   virtual void do_registration_begins () = 0;
+
+  // Override this method to insert logic executed when a pair of source/target
+  // field identifiers are registered.
   virtual void do_register_field (const identifier_type& src, const identifier_type& tgt) = 0;
+
+  // Override this method to insert logic executed when a pair of source/target
+  // fields are bound to an index within the remapper.
   virtual void do_bind_field (const int ifield, const field_type& src, const field_type& tgt) = 0;
+
+  // Override this method to insert logic executed when a field is unregistered.
   virtual void do_unregister_field (const int ifield) = 0;
+
+  // Override this method to insert logic executed when the field registration
+  // process ends.
   virtual void do_registration_ends () = 0;
 
+  // Override this method to implement the forward remapping process using
+  // the protected data members of this class.
   virtual void do_remap_fwd () const = 0;
+
+  // Override this method to implement the backward/inverse remapping process
+  // using the protected data members of this class.
   virtual void do_remap_bwd () const = 0;
 
+  // This helper function searches for a pair of source and target field
+  // identifiers in the remapper's set of bound fields, and returns the index to
+  // which they are bound, or -1 if no such pair of bound fields exists within
```

Comment:
This function returns -1 if fields were not registered at all (via the field identifier). If they were registered, it will return the idx, even if they were not bound yet.